test/test.js

```diff
@@ -37,6 +37,7 @@ function createClient() {
       accessKeyId: process.env.S3_KEY,
       secretAccessKey: process.env.S3_SECRET,
       endpoint: process.env.S3_ENDPOINT,
```

Comment:
@pennycoder here is where the S3_ENDPOINT var is used.

Had to disable ssl for the mock - this should be converted to an env var as well, so current s3 behaviour is preserved as you mentioned.